lib/net.cpp

```diff
@@ -15,6 +15,26 @@
 
 // https://wiki.openssl.org/index.php/SSL/TLS_Client
 
+static std::string socket_error(int err)
+{
+    static char msgbuf[256] = { '\0' };
+#ifdef _WIN32
+    FormatMessage(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+                    NULL,
+                    err,
+                    MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+                    msgbuf,
+                    sizeof(msgbuf),
+                    NULL);
+#else
+    snprintf(msgbuf, sizeof(msgbuf), "%s", strerror(err));
```

Comment:
To avoid an unnecessary copy, this case can be simply `return strerror(err)`. You should be aware of default constructors and how it is unnecessary to explicitly call `std::string(...)` for character buffers.

Reply:
Fixed.